src/string.js

```diff
@@ -77,6 +77,7 @@ inherits(StringSchema, MixedSchema, {
 
     if (options) {
       if (typeof options === 'string') message = options;
+      if (typeof options === 'function') message = options;
```

Comment:
I think this whole inner branch can be rewritten as 
```js
if (typeof options === 'object') {
  ({ excludeEmptyString, message, name } = options);
} else {
  message = options
}
```

Reply:
Thanks for the review, I had defaulted to checking it was a function as there was type checking for strings. The docs do show the supported types so it's really down to user error if they pass an incompatible type

Reply:
in practice, messages can be any type! The string check seems like a bug

Reply:
Perfect, have pushed the update, thanks again!